tests/unit/parts/plugins/test_ubuntu_kernel_plugin.py

```diff
@@ -0,0 +1,985 @@
+            kernel_defconfig=kernel_defconfig,
+            kernel_config_fragments=kernel_config_fragments,
+            kernel_tools=kernel_tools,
+            kernel_dkms_modules=kernel_dkms_modules,
+            kernel_image_target=kernel_image_target,
+        )
+    )
+
+    dpkg_deb_linux_image_cmd = (
+        [
+            f"dpkg-deb -R linux-image-5.15.0-143-generic_5.15.0-143.153_{target_arch}.deb unpacked-linux-image",
+        ]
+        if build_from_binary_package
+        else [
+            f"dpkg-deb -R linux-image-unsigned-5.15.0-143-generic_5.15.0-143.153_{target_arch}.deb unpacked-linux-image",
+        ]
+    )
+    cmds = (
+        [
+            "env",
+            "cd ;parts/ubuntu-kernel/build",
+            "rm -fr ;parts/ubuntu-kernel/build/kernel/",
+            "rm -fr ;parts/ubuntu-kernel/build/unpacked-*",
+            "rm -fr ;parts/ubuntu-kernel/install/*",
+            "mkdir ;parts/ubuntu-kernel/build/kernel",
+        ]
+        + build_type_cmds
+        + [
+            "cd ;parts/ubuntu-kernel/build/kernel",
+        ]
+        + dpkg_deb_linux_image_cmd
+        + [
+            f"dpkg-deb -x linux-modules-5.15.0-143-generic_5.15.0-143.153_{target_arch}.deb unpacked-linux-modules",
+            f"dpkg-deb -x linux-modules-extra-5.15.0-143-generic_5.15.0-143.153_{target_arch}.deb unpacked-linux-modules",
+            "mv unpacked-linux-image/* ;parts/ubuntu-kernel/install",
+            "mkdir ;parts/ubuntu-kernel/install/lib",
+            "mv unpacked-linux-modules/lib/modules ;parts/ubuntu-kernel/install/lib/",
+            "mv unpacked-linux-modules/boot/* ;parts/ubuntu-kernel/install/boot/",
+            "depmod -b ;parts/ubuntu-kernel/install 5.15.0-143-generic",
+            "mv ;parts/ubuntu-kernel/install/boot/* ;parts/ubuntu-kernel/install/",
+            "ln -f ;parts/ubuntu-kernel/install/vmlinuz-5.15.0-143-generic ;parts/ubuntu-kernel/install/kernel.img",
+            "mv ;parts/ubuntu-kernel/install/lib/modules ;parts/ubuntu-kernel/install/",
+            "DTBS=unpacked-linux-firmware/lib/firmware/5.15.0-143/device-tree",
+            '[ -d "${DTBS}" ] && mv "${DTBS}" ;parts/ubuntu-kernel/install/dtbs',
+            "FIRMWARE=unpacked-linux-firmware/lib/firmware/5.15.0-143",
+            '[ -d "${FIRMWARE}" ] && mv "${FIRMWARE}" ;parts/ubuntu-kernel/install/firmware',
+            "rm -rf ;parts/ubuntu-kernel/install/boot",
+        ]
+    )
+    cmd_list = [x.strip() for x in cmds if x]
+    return cmd_list
+
+
+@dataclasses.dataclass
+class BuildParameters:
+    """Parameters for the build normally provided by snapcraft."""
+
+    def __init__(
+        self,
+        base: str,
+        arch_build_on: str,
+        arch_build_for: str,
+        arch_triplet_build_for: str,
+    ):
+        self.base = base
+        self.arch_build_for = arch_build_for
+        self.arch_build_on = arch_build_on
+        self.arch_triplet_build_for = arch_triplet_build_for
```

Comment:
```suggestion
   base: str
   arch_build_for: str
   arch_build_on: str
   arch_triplet_build_for: str
```

You can just use `dataclass`' default init here.

Reply:
done